src/GoTrueClient.ts

```diff
@@ -1562,6 +1562,100 @@ export default class GoTrueClient {
     }
   }
 
+  /**
+   * Gets all the identities linked to a user.
+   */
+  async getUserIdentities(): Promise<
+    | {
+        data: {
+          identities: UserIdentity[]
+        }
+        error: null
+      }
+    | { data: null; error: AuthError }
+  > {
+    try {
+      const { data, error } = await this.getUser()
+      if (error) throw error
+      return { data: { identities: data.user.identities ?? [] }, error: null }
+    } catch (error) {
+      if (isAuthError(error)) {
+        return { data: null, error }
+      }
+      throw error
+    }
+  }
+  /**
+   * Links an oauth identity to an existing user.
+   * This method supports the PKCE flow.
+   */
+  async linkIdentity(credentials: SignInWithOAuthCredentials): Promise<OAuthResponse> {
+    try {
+      const { data, error } = await this._useSession(async (result) => {
+        const { data, error } = result
+        if (error) throw error
+        const url: string = await this._getUrlForProvider(
+          `${this.url}/user/identities/authorize`,
+          credentials.provider,
+          {
+            redirectTo: credentials.options?.redirectTo,
+            scopes: credentials.options?.scopes,
+            queryParams: credentials.options?.queryParams,
+            skipBrowserRedirect: true,
+          }
+        )
+        return await _request(this.fetch, 'GET', url, {
+          headers: this.headers,
+          jwt: data.session?.access_token ?? undefined,
+        })
+      })
+      if (error) throw error
+      if (isBrowser() && !credentials.options?.skipBrowserRedirect) {
+        window.location.assign(data?.url)
+      }
+      return { data: { provider: credentials.provider, url: data?.url }, error: null }
+    } catch (error) {
+      if (isAuthError(error)) {
+        return { data: { provider: credentials.provider, url: null }, error }
+      }
+      throw error
+    }
+  }
+
+  /**
+   * Unlinks an identity from a user by deleting it. The user will no longer be able to sign in with that identity once it's unlinked.
+   */
+  async unlinkIdentity(identity: UserIdentity): Promise<
+    | {
+        data: {}
```

Comment:
decided to use an empty object instead so that it is consistent with the return types of other methods like `resetPasswordForEmail` and `signInWithOtp`